flexmeasures/data/models/planning/utils.py

```diff
@@ -441,3 +442,14 @@ def nanmin_of_series_and_value(s: pd.Series, value: float | pd.Series) -> pd.Ser
         # [right]: datetime64[ns, UTC]
         value = value.tz_convert("UTC")
     return s.fillna(value).clip(upper=value)
+
+
+def get_soc_sensor_value(sensor: Sensor, start: datetime):
```

Comment:
I think we'll need some sort of inference here, e.g. in case the last SoC value was a couple of minutes before `start`. One possibility: something like `sensor.search_beliefs(most_recent_event=True, events_before=start)`. Possibly with a `events_after=start-max_staleness` is we don't want the last SoC value to be unacceptably old.

Note that I consider this a comment for the follow-up PR, and I consider this function out of scope for this PR, which should be just about saving the SoC values to the sensor, not already getting the SoC values from the sensor.

Reply:
what should be the value of `max-staleness`?

Reply:
Again, I consider this a comment for a follow-up PR/Issue, to be discussed there. And (there) I'd leave it None for now, in which case the filter for `events_after' can be left out. Just copy or link to this comment thread on the new Issue, so we can discuss where this max_staleness parameter should live.